classes/local/store/object_file_system.php

```diff
@@ -756,6 +762,29 @@ public function generate_presigned_url_to_external_file($contenthash, $headers =
         return $this->externalclient->generate_presigned_url($contenthash, $headers);
     }
 
+    /**
+     * Check if file extension is whitelisted.
+     * @param string $filename
+     * @return bool
+     * @throws \dml_exception
+     */
+    public function is_extension_whitelisted($filename) {
+        $config = manager::get_objectfs_config();
+        if (empty($config->signingwhitelist)) {
+            return false;
+        }
+        $whitelist = explode(',',  $config->signingwhitelist);
```

Comment:
just be careful with whitespace, this can be edited by humans directly and whitespace around each item should be trimmed